### PR TITLE
executor/inspection_common_test.go: migrate test-infra

### LINKED ISSUE #28589: migrate test-infra to testify for `inspection_common_test.go`

### PR DESCRIPTION
**Pull request by bodhisatan**
### What problem does this PR solve?

Issue Number: close #28589 
Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/inspection_common_test.go [hunk 1]
```diff
@@ -52,10 +53,10 @@ func (s *inspectionSummarySuite) TestInspectionRules(c *C) {
...
 		rs, err := tk.Exec(ca.sql)
-		c.Assert(err, IsNil)
-		rules, err := session.ResultSetToStringSlice(context.Background(), tk.Se, rs)
-		c.Assert(err, IsNil)
-		c.Assert(len(rules), Equals, ca.ruleCount)
-		c.Assert(rs.Close(), IsNil)
+		require.NoError(t, err)
+		rules, err := session.ResultSetToStringSlice(context.Background(), tk.Session(), rs)
+		require.NoError(t, err)
+		require.Equal(t, len(rules), ca.ruleCount)
```
Comment (hawkingrei) [inline executor/inspection_common_test.go:59]: require.Len(t, rules, ca.rule.Count)

**Comment by hawkingrei** [comment]
Please fix the linter.

```
[2021-12-16T15:56:06.483Z] level=info msg="[runner] linters took 54.031051564s with stages: goanalysis_metalinter: 53.747208204s"

[2021-12-16T15:56:06.483Z] executor/inspection_common_test.go:19: File is not `goimports`-ed (goimports)

[2021-12-16T15:56:06.483Z] 	"github.com/stretchr/testify/require"

[2021-12-16T15:56:06.483Z] level=info msg="File cache stats: 1329 entries of total size 22.0MiB"

[2021-12-16T15:56:06.483Z] level=info msg="Memory: 1376 samples, avg is 1832.0MB, max is 9451.8MB"

[2021-12-16T15:56:06.483Z] level=info msg="Execution took 2m24.630377838s"

```

**Comment by bodhisatan** [comment]
> Please fix the linter.
> 
> ```
> [2021-12-16T15:56:06.483Z] level=info msg="[runner] linters took 54.031051564s with stages: goanalysis_metalinter: 53.747208204s"
> 
> [2021-12-16T15:56:06.483Z] executor/inspection_common_test.go:19: File is not `goimports`-ed (goimports)
> 
> [2021-12-16T15:56:06.483Z] 	"github.com/stretchr/testify/require"
> 
> [2021-12-16T15:56:06.483Z] level=info msg="File cache stats: 1329 entries of total size 22.0MiB"
> 
> [2021-12-16T15:56:06.483Z] level=info msg="Memory: 1376 samples, avg is 1832.0MB, max is 9451.8MB"
> 
> [2021-12-16T15:56:06.483Z] level=info msg="Execution took 2m24.630377838s"
> ```

fixed :)

**Comment by bodhisatan** [comment]
@hawkingrei 3q~

**Review by tangenta — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM! Sorry for noticing this PR late.

**Review by hawkingrei — APPROVED** [review]

File: executor/inspection_common_test.go [hunk 2]
```diff
@@ -16,15 +16,17 @@ package executor_test
...
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/executor"
 	"github.com/pingcap/tidb/session"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/testkit"
 )
 
-func (s *inspectionSummarySuite) TestInspectionRules(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func (s *inspectionSummarySuite) TestInspectionRules(t *testing.T) {
```
Comment (hawkingrei) [inline executor/inspection_common_test.go:28]: Please remove ```(s *inspectionSummarySuite)```

Comment (hawkingrei) [reply executor/inspection_common_test.go:28]: Ditto other.

Comment (tisonkun) [reply executor/inspection_common_test.go:28]:
```suggestion
func TestInspectionRules(t *testing.T) {
```

**Pull request merged**